### PR TITLE
[VL] Add plan validation util for debugging validate process

### PR DESCRIPTION
**Pull request by Yohahaha**
## What changes were proposed in this pull request?

Add plan_validation_util executable for debugging validate process in native side, and refine docs.

### CODE REVIEW

**Comment by Yohahaha** [comment]
@marin-ma @PHILO-HE please help to review this patch, thank you!

File: cpp/velox/jni/VeloxJniWrapper.cc [hunk 1]
```diff
@@ -92,11 +94,21 @@ JNIEXPORT void JNICALL Java_io_glutenproject_udf_UdfJniWrapper_nativeLoadUdfLibr
...
   JNI_METHOD_START
+  auto ctx = gluten::getRuntime(env, wrapper);
   auto planData = reinterpret_cast<const uint8_t*>(env->GetByteArrayElements(planArray, 0));
   auto planSize = env->GetArrayLength(planArray);
+  if (gluten::debugModeEnabled(ctx->getConfMap())) {
+    try {
+      auto jsonPlan = gluten::substraitFromPbToJson("Plan", planData, planSize);
+      LOG(INFO) << std::string(50, '#') << " received substrait::Plan: for validation";
+    } catch (const std::exception& e) {
+      LOG(WARNING) << "Error converting Substrait plan to JSON: " << e.what();
```
Comment (marin-ma) [inline cpp/velox/jni/VeloxJniWrapper.cc:108]: Probably "Error converting Substrait plan for validation to JSON:"

File: docs/developers/MicroBenchmarks.md [hunk 2]
```diff
@@ -94,23 +94,14 @@ InputFromBatchVector/iterations:1/process_time/real_time/threads:1   41304520 ns
...
 
-First, rebuild Gluten with build type `RelWithDebInfo`.
-
-```shell
-cd /path_to_gluten/
-./dev/buildbundle-veloxbe.sh --build_tests=ON --build_benchmarks=ON --build_type=RelWithDebInfo
-
-```
-Run the query by spark-shell.  Get the Substrait plan from executor's stdout.
+Run the query with config `spark.gluten.sql.debug=true` and get the Substrait json plan from log, then save it as a JSON file, suppose the name is "plan.json".
```
Comment (marin-ma) [inline docs/developers/MicroBenchmarks.md:97]: We also need config `spark.gluten.sql.columnar.backend.velox.glogSeverityLevel=0`

Comment (Yohahaha) [reply docs/developers/MicroBenchmarks.md:97]: thank you, btw could we open glog when debug mode enabled?

**Comment by marin-ma** [comment]
Thanks for this feature! Could you provide some output after this change?
I'm not sure that all validation plans can be successfully logged. Could you check if we need to apply this fix to the validation plan on java side?
https://github.com/oap-project/gluten/pull/3969/files#diff-a7356fa0629d859807e18644ff6556e6babc825fc25c95a967ce28abe6d100d9R577-R578

File: docs/developers/MicroBenchmarks.md (hunk 2 above)
Comment (marin-ma) [reply docs/developers/MicroBenchmarks.md:97]: > thank you, btw could we open glog when debug mode enabled?

Sounds good!

**Comment by Yohahaha** [comment]
> Thanks for adding this feature! Could you provide some output after this change? I'm not sure that all validation plans can be successfully logged. Could you check if we need to apply this fix to the validation plan on java side? https://github.com/oap-project/gluten/pull/3969/files#diff-a7356fa0629d859807e18644ff6556e6babc825fc25c95a967ce28abe6d100d9R577-R578

json plan can be printed if has no agg. your patch is great, I will revert doc modification.

File: cpp/velox/jni/VeloxJniWrapper.cc (hunk 1 above)
Comment (Yohahaha) [reply cpp/velox/jni/VeloxJniWrapper.cc]: done.

**Comment by Yohahaha** [comment]
```
         > select count(*) from test;
I1212 10:35:08.637563 3619404 VeloxJniWrapper.cc:98] ################################################## received substrait::Plan: for validation
I1212 10:35:08.637595 3619404 VeloxJniWrapper.cc:99] {"relations":[{"root":{"input":{"read":{"common":{"direct":{}},"baseSchema":{"struct":{}}}}}}]}
[libprotobuf WARNING google/protobuf/util/internal/json_objectwriter.cc:52] JsonObjectWriter was not fully closed.
W1212 10:35:08.784191 3619404 VeloxJniWrapper.cc:101] Error converting Substrait plan for validation to JSON: BinaryToJsonStream returned INTERNAL:Invalid type URL, type URLs must be of the form '/<typename>', got: type.googleapis.com/google.protobuf.StringValue



         > select * from test;
I1212 10:35:41.984709 3619404 VeloxJniWrapper.cc:98] ################################################## received substrait::Plan: for validation
I1212 10:35:41.984735 3619404 VeloxJniWrapper.cc:99] {"relations":[{"root":{"input":{"read":{"common":{"direct":{}},"baseSchema":{"names":["id","b"],"struct":{"types":[{"i32":{"nullability":"NULLABILITY_NULLABLE"}},{"bool":{"nullability":"NULLABILITY_NULLABLE"}}]},"columnTypes":["NORMAL_COL","NORMAL_COL"]}}}}}]}

```

**Comment by Yohahaha** [comment]
@marin-ma could we merge or any more comments?

**Comment by PHILO-HE** [comment]
Hi @Yohahaha, could you document how to use it? Maybe, in `HowTo.md`. Thanks!

**Comment by Yohahaha** [comment]
> Hi @Yohahaha, could you document how to use it? Maybe, in `HowTo.md`. Thanks!

Added.

**Review by PHILO-HE — APPROVED** [review]
Thanks for your work! Will merge the PR soon (Rong is taking leave, if any more comment after that, let's fix it in follow-up PR).

**Pull request merged**